src/main/java/gregtech/common/covers/detector/CoverDetectorFluidAdvanced.java

```diff
@@ -0,0 +1,172 @@
+                this::getMinValue, this::setMinValue)
+                .setMaxLength(10)
+                .setAllowedChars(Pattern.compile(".[0-9]*"))
+                .setPostFix("L")
+        );
+
+        // set max fluid amount
+        group.addWidget(new LabelWidget(10, 5 + 2 * (SIZE + PADDING), "cover.advanced_fluid_detector.max"));
+        group.addWidget(new ImageWidget(98 - 4, 2 * (SIZE + PADDING), 4 * SIZE, SIZE, GuiTextures.DISPLAY));
+        group.addWidget(new TextFieldWidget2(98, 5 + 2 * (SIZE + PADDING), 4 * SIZE, SIZE,
+                this::getMaxValue, this::setMaxValue)
+                .setMaxLength(10)
+                .setAllowedChars(Pattern.compile(".[0-9]*"))
+                .setPostFix("L")
+        );
+
+        // invert logic button
+        group.addWidget(new LabelWidget(10, 5 + 3 * (SIZE + PADDING), "cover.advanced_energy_detector.invert_label"));
+        group.addWidget(new CycleButtonWidget(98 - 4, 3 * (SIZE + PADDING), 4 * SIZE, SIZE, this::isInverted, this::setInverted,
+                "cover.advanced_energy_detector.normal", "cover.advanced_energy_detector.inverted")
+                .setTooltipHoverString("cover.advanced_fluid_detector.invert_tooltip")
+        );
+
+        this.fluidFilter.initUI(5 + 4 * (SIZE + PADDING), group::addWidget);
+
+        return ModularUI.builder(GuiTextures.BACKGROUND,  176, 164 + 82)
+                .widget(group)
+                .bindPlayerInventory(player.inventory, GuiTextures.SLOT, 7, 164)
+                .build(this, player);
+    }
+
+    private String getMinValue() {
+        return String.valueOf(min);
+    }
+    private String getMaxValue() {
+        return String.valueOf(max);
+    }
+    private void setMinValue(String val){
+        int parsedValue = GTUtility.tryParseInt(val, DEFAULT_MIN);
+        this.min = Math.min(max - 1, Math.max(0, parsedValue));
+    }
+    private void setMaxValue(String val){
+        int parsedValue = GTUtility.tryParseInt(val, DEFAULT_MAX);
+        this.max = Math.max(min + 1, parsedValue);
+    }
+    private boolean isInverted(){
+        return this.isInverted;
+    }
+    private void setInverted(boolean b){
+        this.isInverted = b;
+    }
+    @Override
+    public void update() {
+        if (this.coverHolder.getOffsetTimer() % 20 != 0)
+            return;
+
+        IFluidHandler fluidHandler = coverHolder.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, null);
+        if (fluidHandler == null)
+            return;
+
+        IFluidTankProperties[] tankProperties = fluidHandler.getTankProperties();
+        int storedFluid = 0;
+
+        for (IFluidTankProperties properties : tankProperties) {
+            FluidStack contents = properties.getContents();
+
+            if (contents != null && fluidFilter.testFluidStack(contents))
+                storedFluid += contents.amount;
+        }
+
+        setRedstoneSignalOutput(GTUtility.compareValue(storedFluid, max, min, isInverted));
+    }
```

Comment:
These methods could use a blank line between them, and a space before the opening curly brace, for codestyle consistency.

Reply:
should be resolved with https://github.com/GregTechCEu/GregTech/pull/1448/commits/6779bc5d3151a8962d87833a1835ae2ab5571fcf